src/installer/installer.go

```diff
@@ -466,37 +462,38 @@ func (i *installer) waitForController() error {
 		i.log.WithError(err).Errorf("Failed to create kc client from %s", KubeconfigPath)
 		return err
 	}
+
 	events := map[string]string{}
-	predicate := func() bool {
-		controllerPod := common.GetPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
-			map[string]string{"job-name": assistedControllerPrefix}, v1.PodRunning, i.log)
-		if controllerPod != nil {
-
-			// uploading logs here cause we will finish installing bootstrap in couple of seconds
-			// just didn't wanted to start handling this logic in other place
-			// controller must be running for at least couple of minutes before this code so it will cover 90% of the cases
-			// if we will see that it is not enough, we will change the place.
-			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
-			// if failed to upload logs, log why and continue
-			if err != nil {
-				if retryCounterUploadControllerLogs > 0 {
-					retryCounterUploadControllerLogs--
-					controllerPod = nil
+	tickerUploadLogs := time.NewTicker(5 * time.Minute)
+	tickerWaitForController := time.NewTicker(waitForControllerPodInterval)
+	// Keep trying until we're time out or get true
+	for {
+		select {
+		case <-tickerWaitForController.C:
+			i.logControllerEvents(kc, events)
```

Comment:
why does it called logControllerEvents? Doesnt it listen to events?
Cannot u just register to events rather than polling?

Reply:
Agree. you should use a watcher here.

Reply:
we are logging all new events that were added since we want to understand what is going on. As i said in message above, this code will 99% run once, i don't think there is any need to create listener on code that will run for maximum 5 minutes and better to have polling. When this code is running mainly all events will be set already.

Reply:
In any case we will need to poll even if we add listener so for such case it is not worth it, it will only complicate it